python/pyspark/sql/column.py

```diff
@@ -329,6 +329,31 @@ def getField(self, name):
                 DeprecationWarning)
         return self[name]
 
+    @since(3.1)
+    def withField(self, fieldName, col):
+        """
+        An expression that adds/replaces field in StructType by name.
+
+        >>> from pyspark.sql import Row, functions
```

Comment:
I would import `lit` as:

```python
from pyspark.sql.functions import lit
```

Reply:
Done

Reply:
@Kimahriman, seems you missed this (?).

Reply:
Oh woops I did that for the test not for the docstring, sorry

Reply:
Actually done this time